docker-compose.yaml

```diff
@@ -0,0 +1,41 @@
+# The Docker Config version. Do not change.
+version: "3.1"
+services:
+  frontend:
+    build: ./client
+    container_name: client
+    ports:
+      - "3000:3000"
+    networks:
+      - app-network
+    depends_on:
+      - server
+
+  server:
+    build: ./server
+    container_name: server
+    ports:
+      - ${SERVER_PORT}:${SERVER_PORT}
+    networks:
+      - app-network
+    depends_on:
+      - db
+
+  db:
+    image: postgres
+    container_name: db
+    ports:
+      - ${POSTGRES_PORT}:${POSTGRES_PORT}
+    environment:
+      POSTGRES_USER: ${POSTGRES_USER}
+      POSTGRES_PASSWORD: ${POSTGRES_USER_PW}
+      POSTGRES_DB: ${POSTGRES_DB}
+      POSTGRES_HOST: ${POSTGRES_HOST}
+    volumes:
+      - ./my_data_directory:/var/lib/postgresql/data
```

Comment:
Instead of specifying a bind mount for the db persistence, we could default to a normal docker volume. This would prevent us having to add a directory to the .gitignore

```
volumes:
  db-data:
```